### PR TITLE
feat: remove create bucket & object approval

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

Remove the create object approval & create bucket approval, and during the process of creating objects & buckets, the calls to the SP side are eliminated through the Go-sdk.

For CreateObject, validation for the object's existence has been added on chain, and the corresponding verify signature has been removed. 

Virtual Group Manager will automatically check the available status of each SP's GVG & VGF and create GVG when the storage is full or the SP does not have any available GVG.

#### Before 
SP Pick VGF & Create GVG process
image

#### Current
image
Replace the process on the SP side of checking for available VGF & GVG during the create bucket approval.
Virtual Group Manager will automatically check the available status of each SP's GVG & VGF and create GVG when the storage is full or the SP does not have any available GVG.

### Example

N/A

### Changes

Notable changes: 
* Virtual Group Manager will automatically check the available status of each SP's GVG & VGF and create GVG when the storage is full or the SP does not have any available GVG

### Potential Impacts
* Go-SDK

### CODE REVIEW

File: modular/manager/manager.go [hunk 1]
```diff
@@ -1195,3 +1201,25 @@ func (m *ManageModular) QueryRecoverProcess(ctx context.Context, vgfID, gvgID ui
...
+		spList []*sptypes.StorageProvider
+	)
+
+	// query meta
+	if spList, err = m.baseApp.Consensus().ListSPs(context.Background()); err != nil {
+		log.CtxErrorw(ctx, "failed to list sps", "error", err)
+		return
+	}
+
+	for _, sp := range spList {
```
Comment (alexgao001) [inline modular/manager/manager.go:1217]: no need to loop through the SP list

Comment (BarryTong65) [reply modular/manager/manager.go:1217]: Why is there no need to loop through the SP list? I need to ensure that every STATUS_IN_SERVICE SP has available VGF & GVG. If we don't query each SP in the SP lists to determine if a GVG needs to be created, it seems necessary to use a for loop to iterate through them.

Comment (BarryTong65) [reply modular/manager/manager.go]: fixed

File: modular/manager/manager_options.go [hunk 2]
```diff
@@ -87,6 +87,9 @@ const (
 	// on greenfield timeout height, if after current block height + timeout height, the object
 	// is not rejected, it is judged failed to reject unseal object on greenfield.
 	DefaultListenRejectUnSealTimeoutHeight int = 10
+	// DefaultSyncAvailableVGFInterval defines the default interval for available global virtual group family info,
+	// it is used to log and debug.
+	DefaultSyncAvailableVGFInterval int = 60
```
Comment (clydemeng) [inline modular/manager/manager_options.go:92]: seconds, right?

File: modular/manager/manager.go [hunk 3]
```diff
@@ -1195,3 +1201,24 @@ func (m *ManageModular) QueryRecoverProcess(ctx context.Context, vgfID, gvgID ui
...
+		sp  *sptypes.StorageProvider
+	)
+
+	// query meta
+	if sp, err = m.baseApp.Consensus().QuerySPByID(context.Background(), m.spID); err != nil {
+		log.CtxErrorw(ctx, "failed to list sps", "error", err)
+		return
+	}
+
+	// only pick vgf when sp is STATUS_IN_SERVICE
```
Comment (clydemeng) [inline modular/manager/manager.go:1217]: do we need to take GVGPreferSPList and SPBlackList into account when picking vgf?

File: modular/manager/manager_options.go (hunk 2 above)
Comment (BarryTong65) [reply modular/manager/manager_options.go:92]: yes

File: modular/manager/manager.go (hunk 3 above)
Comment (BarryTong65) [reply modular/manager/manager.go:1217]: SPBlackList will be used when picking VGF
image

GVGPreferSPList will be used when creating GVG
image

image

File: base/gfspconfig/config.go [hunk 4]
```diff
@@ -222,6 +222,8 @@ type ParallelConfig struct {
 
 	LoadReplicateTimeout int64 `comment:"optional"`
 	LoadSealTimeout      int64 `comment:"optional"`
+
+	SyncAvailableVGFInterval int `comment:"optional"`
```
Comment (KeefeL) [inline base/gfspconfig/config.go:226]: You can directly hardcode it in the code using macros.
Exporting as a config will increase the cognitive and maintenance costs of the SP.

Comment (BarryTong65) [reply base/gfspconfig/config.go]: fixed with default value 5 mins

**Review by ruojunm — APPROVED** [review]
LGTM

**Review by KeefeL — APPROVED** [review]

**Pull request merged**